### PR TITLE
✨ feat: Add 'minor' field to Model schema and integrate into model details editor and generation service

### PR DESCRIPTION
**Pull request by bryantanjw**
### What's changed
- Added `minor` field to the Model schema in prisma.
- Updated the `ModelUpsertForm` to include a checkbox for `minor`.
- Added validation to prevent NSFW content depicting minors.
- Updated `generationResourceSelect` to include `minor` field.
- Modified `createGenerationRequest` to disable NSFW if any resource is marked as `minor`.
- Updated model controller to handle `minor` field.
- Adjusted zod schemas to include `minor` field.

### CODE REVIEW

File: src/server/services/generation/generation.service.ts [hunk 1]
```diff
@@ -940,6 +945,7 @@ const getImageGenerationData = async (id: number): Promise<Generation.Data> => {
       m.id "modelId",
       m.name "modelName",
       m.type "modelType",
+      m.minor,
```
Comment (manuelurenah) [inline src/server/services/generation/generation.service.ts:948]: I don't think we need this one here, can be safely removed
```suggestion
```

**Review by manuelurenah — APPROVED** [review]
Sweet 🔥

File: src/server/services/generation/generation.service.ts (hunk 1 above)
Comment (bryantanjw) [reply src/server/services/generation/generation.service.ts]: ahh, good catch

**Review by lrojas94 — APPROVED** [review]

File: src/pages/api/webhooks/image-scan-result.ts [hunk 2]
```diff
@@ -280,15 +280,16 @@ async function handleSuccess({ id, tags: incomingTags = [], source, context }: B
...
           -- Check based on associated resources
           SELECT
-            COUNT(*) > 0 "poi"
+            COUNT(*) > 0 "poi",
+            COUNT(*) FILTER (WHERE m.minor) > 0 "minor"
           FROM "ImageResource" ir
           JOIN "ModelVersion" mv ON ir."modelVersionId" = mv.id
           JOIN "Model" m ON m.id = mv."modelId"
-          WHERE ir."imageId" = ${image.id} AND m.poi
+          WHERE ir."imageId" = ${image.id}
```
Comment (manuelurenah) [inline src/pages/api/webhooks/image-scan-result.ts:292]: If we remove the `AND` here, I guess it'll count all resources as POI 🤔 we probably need a filter for "poi" as well

**Review by manuelurenah — APPROVED** [review]

**Pull request merged**